src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -272,6 +280,22 @@ public void firePostConstructEvents() {
         this.isConstructing = false;
     }
 
+    @Inject(method = "addPassenger(Lnet/minecraft/entity/Entity;)V", at = @At("HEAD"), cancellable = true)
+    public void onAddPassenger(net.minecraft.entity.Entity passenger, CallbackInfo ci) {
+        if (!this.worldObj.isRemote) {
+            if (SpongeImpl.postEvent(SpongeEventFactory.createMountEntityEvent(Cause.of(NamedCause.source(passenger)), this))) {
```

Comment:
You can also add to `org.spongepowered.common.event.ShouldFire` for whether plugins are listening to this event, and if it is false, don't bother throwing the event ;)